tests/simulationbase.py

```diff
@@ -39,8 +42,18 @@ def __init__(self, work_dir: Path) -> None:
     def init_provider(
             self,
             get_task_api_service: Callable[[Path], TaskApiService],
+            task_id
     ) -> None:
-        self.get_provider_task_api_service = get_task_api_service
+        self.mkdir_provider_task(task_id)
+        self._task_api_service = get_task_api_service(self.prov_task_work_dir)
+
+    @asynccontextmanager
+    async def start_provider(self) -> None:
+        self.provider_client = ProviderAppClient(self._task_api_service)
+        try:
+            yield self.provider_client
+        finally:
+            await self.provider_client.shutdown()
```

Comment:
After a successful `compute/benchmark` one does not need to call shutdown, it'll shutdown automatically.

Reply:
Agree in cases where the test goes well, but when it raises an error it could leave the app running.

Can add a comment that this is not required but nice to have for cleanup or remove the line, what do you think is best?

Reply:
Good point, I think the proper way would be to put these lines https://github.com/golemfactory/task-api/blob/master/python/golem_task_api/client.py#L163-L164 into `try/catch` block to always shutdown even when an error occurs. Same in the benchmark method.

Reply:
haha, just removed that try/catch block since it did nothing :) will add it back with this safety shutdown